### ISSUE TITLE
Poetry always reinstalls all packages (poetry add, poetry update) on windows 10

### ISSUE THREAD

**Issue by AlxndrJhn** [python-poetry/poetry#2677]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: windows 10 ,version 1909
- **Poetry version**: Poetry version 1.0.9
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/AlxndrJhn/b59c7426d3264159318c7fd04364276f

## Issue
Just like in #2502 poetry re-installs all packages every time I update anything, it re-installs the same version.
I created the virtual environment with `poetry shell`.
As suggested, I already deleted the lock file and re-created it with `poetry lock`

The terminal log below shows that the `update` command was given a single package from a private repo (that is defined in toml file, see above) to update, but re-installed all packages unnecessarily.
The same happens when I add a new package with `poetry add` and update everything `poetry update`.
I had this issue since I started using poetry on windows (since version 1.0.5).

```text
(us-tools-XDCreN_q-py3.8) PS C:\Users\jahna\UltrasoundMeasurementCell\us-tools> python --version
Python 3.8.0
(us-tools-XDCreN_q-py3.8) PS C:\Users\jahna\UltrasoundMeasurementCell\us-tools> pip --version
pip 19.2.3 from c:\users\jahna\appdata\local\pypoetry\cache\virtualenvs\us-tools-xdcren_q-py3.8\lib\site-packages\pip (python 3.8)
(us-tools-XDCreN_q-py3.8) PS C:\Users\jahna\UltrasoundMeasurementCell\us-tools> which python

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     python.exe                                         3.8.150... C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\Scripts\python.exe


(us-tools-XDCreN_q-py3.8) PS C:\Users\jahna\UltrasoundMeasurementCell\us-tools> which pip

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     pip.exe                                            0.0.0.0    C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\Scripts\pip.exe


(us-tools-XDCreN_q-py3.8) PS C:\Users\jahna\UltrasoundMeasurementCell\us-tools> which poetry

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     poetry.bat                                         0.0.0.0    C:\Users\jahna\.poetry\bin\poetry.bat


(us-tools-XDCreN_q-py3.8) PS C:\Users\jahna\UltrasoundMeasurementCell\us-tools> poetry update ultrasoundalgorithm
Updating dependencies
Resolving dependencies...


Package operations: 0 installs, 45 updates, 0 removals

  - Updating certifi (2020.6.20 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 2020.6.20)
  - Updating chardet (3.0.4 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 3.0.4)
  - Updating click (7.1.2 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 7.1.2)
  - Updating dnspython (1.16.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.16.0)
  - Updating h11 (0.9.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.9.0)
  - Updating idna (2.10 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 2.10)
  - Updating mypy-extensions (0.4.3 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.4.3)
  - Updating pydantic (1.6.1 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.6.1)
  - Updating pyparsing (2.4.7 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 2.4.7)
  - Updating six (1.15.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.15.0)
  - Updating starlette (0.12.9 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.12.9)
  - Updating typed-ast (1.4.1 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.4.1)
  - Updating typing-extensions (3.7.4.2 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 3.7.4.2)
  - Updating urllib3 (1.25.9 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.25.9)
  - Updating websockets (8.1 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 8.1)
  - Updating appdirs (1.4.4 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.4.4)
  - Updating atomicwrites (1.4.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.4.0)
  - Updating attrs (19.3.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 19.3.0)
  - Updating colorama (0.4.3 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.4.3)
  - Updating cycler (0.10.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.10.0)
  - Updating email-validator (1.1.1 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.1.1)
  - Updating et-xmlfile (1.0.1 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.0.1)
  - Updating fastapi (0.46.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.46.0)
  - Updating jdcal (1.4.1 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.4.1)
  - Updating kiwisolver (1.2.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.2.0)
  - Updating more-itertools (8.4.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 8.4.0)
  - Updating mypy (0.761 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.761)
  - Updating numpy (1.19.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.19.0)
  - Updating packaging (20.4 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 20.4)
  - Updating pathspec (0.8.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.8.0)
  - Updating pillow (7.2.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 7.2.0)
  - Updating pluggy (0.13.1 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.13.1)
  - Updating py (1.9.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 1.9.0)
  - Updating python-dateutil (2.8.1 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 2.8.1)
  - Updating regex (2020.7.14 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 2020.7.14)
  - Updating requests (2.24.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 2.24.0)
  - Updating toml (0.10.1 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.10.1)
  - Updating uvicorn (0.11.5 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.11.5)
  - Updating wcwidth (0.2.5 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 0.2.5)
  - Updating black (19.10b0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 19.10b0)
  - Updating matplotlib (3.2.2 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 3.2.2)
  - Updating openpyxl (3.0.4 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 3.0.4)
  - Updating pytest (5.4.3 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 5.4.3)
  - Updating tqdm (4.47.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 4.47.0)
  - Updating ujson (3.0.0 C:\Users\jahna\AppData\Local\pypoetry\Cache\virtualenvs\us-tools-XDCreN_q-py3.8\lib\site-packages -> 3.0.0)
```

I executed the same command again but with `-vvv` flag:
[verbose_update.txt](https://github.com/python-poetry/poetry/files/4931465/verbose_update.txt)

**Comment by abn** [comment]
@AlxndrJhn can you please try with the latest pre-release version?

**Comment by AlxndrJhn** [comment]
fixed in Poetry version 1.1.0a3
```bash
PS C:\Users\jahna\UltrasoundMeasurementCell\us-tools> poetry --version
Poetry version 1.1.0a3
PS C:\Users\jahna\UltrasoundMeasurementCell\us-tools> poetry update ultrasoundalgorithm
Updating dependencies
Resolving dependencies...

Writing lock file

No dependencies to install or update
# I pushed a version bump in the other repository
PS C:\Users\jahna\UltrasoundMeasurementCell\us-tools> poetry update ultrasoundalgorithm
Updating dependencies
Resolving dependencies...

Writing lock file

Package operations: 0 installs, 1 update, 0 removals

  • Updating ultrasoundalgorithm (0.1.3 49b18aa -> 0.1.4 743a1ad)
```
